contents/data/nosql/NoSQL.md

```diff
@@ -0,0 +1,437 @@
+  * [Not Suited for Complex Relational Queries‎](#2-not-suited-for-complex-relational-queries)  
+* [How to get Started With NoSQL?‎](#how-to-get-started-with-nosql)
+
+{.px-3}
+
+</box>
+
+---
+
+## What is NoSQL?
+
+**Non-Structured Query Language** (**NoSQL**) is a wide set of implementations of query technologies used to retrieve and store data in a non-tabular format.
+
+For example, here is one such data item - a book in the catalogue of an e-commerce website, reprsented in a
+<popover effect="fade" header="Javascript Object Notation" content="A commonly used data format which represents data in a simple, human and machine-readable format." placement="top">JSON</popover>
+format.
+ 
+```js
+{
+  type: "book",
+  price: 20,
+  popularity: 9.7,
+}
+```
+
+To start, let's jump into the most common implementation of such databases, _document databases_.
+
+<box type="info" light>
+<li>Other types of NoSQL databases, such as <trigger for="key-value-db">key-value databases</trigger>,
+<trigger for="graph-db">graph databases</trigger>, can be viewed as extensions or reductions of document based databases.
+</li>
+<li>Some of these types can even be used together!</li>
+</box>
+
+One such commonly used database is MongoDB, which is a document based database.
+
+
+![](documentDatabaseStructure.png)
+
+In contrast to tables and table entries in SQL databases, document dtabases comprise of collections and documents.
+A database comprises of multiple collections, which in turn consists of multiple documents.
+
+In a simplified e-commerce website for example, you may have the following collections: { id="main-example" }
+- _customers_ - storing the account details of customers, their purchase histories, etc.
+- _items_ - a collection of all items available for purchase (which are documents)
+- _admin_ - a collection storing admin account details
+- **...**
+
+In this case, the _items_ collection which contains the catalogue of purchase items may be structured like so:
+
+<div id="schemaless-example">
+
+```js
+[
+  {
+    type: "book",
+    title: "about pandas",
+    price: 20,
+    popularity: 9.7,
+    author: "panda1",
+    ...
+  },
+  {
+    type: "grocery",
+    name: "cheese",
+    brand: "panda",
+  },
+  ...
+]
+```
+
+</div>
+
+<br>
+<box type="info" light>
+Note that while many NoSQL databases provide a JSON interface to interact with the data,
+the underlying storage implementation may be different for performance reasons.<br>
+For example, MongoDB stores documents in BSON (json encoded in binary). 
+</box>
+
+<br>
+
+## Key Characteristics of NoSQL Databases
+
+### 1. Powerful and Simple CRUD Operations
+
+Interaction with most document databases is achieved in a simple and intuitive _object oriented manner_,
+and JSON-like queries.
+
+Let's get back to the above [example](#main-example) of an e-commerce website. To register a new user account,
+an example query made in mongoDB would be like so:
```

Comment:
The word _query_ feels like a read operation. Are write operations also called queries?